jubilant/_juju.py

```diff
@@ -119,6 +119,41 @@ def add_model(
         self.cli(*args, include_model=False)
         self.model = model
 
+    def add_unit(
+        self,
+        app: str,
+        *,
+        attach_storage: str | Iterable[str] | None = None,
+        num_units: int = 1,
+        to: str | Iterable[str] | None = None,
+    ):
+        """Add one or more units to a deployed application.
+
+        Args:
+            app: Name of application to add units to.
+            attach_storage: Existing storage(s) to attach to the deployed unit, for example,
+                ``foo/0`` or ``mydisk/1``. Not available for Kubernetes models.
+            num_units: Number of units to add.
+            to: Machine or container to deploy the unit in (bypasses constraints). For example,
+                to deploy to a new LXD container on machine 25, use ``lxd:25``.
+        """
+        args = ['add-unit', app]
+
+        if attach_storage:
+            if isinstance(attach_storage, str):
+                args.extend(['--attach-storage', attach_storage])
+            else:
+                args.extend(['--attach-storage', ','.join(attach_storage)])
+        if num_units != 1:
+            args.extend(['--num-units', str(num_units)])
```

Comment:
Would it make sense to validate the range, like disallowing negative value and zero?
Or does the user get a reasonable error from Juju cli in this case?

Reply:
Yep, the Juju CLI gives a nice message -- in general, I'm relying on the CLI for validation of most of these things.

```
$ juju add-unit database --num-units -3
ERROR --num-units must be a positive integer
```